torch/csrc/Generator.h

```diff
@@ -14,16 +14,16 @@ struct THPGenerator {
 // Creates a new Python object wrapping the default at::Generator. The reference is
 // borrowed. The caller should ensure that the at::Generator object lifetime
 // last at least as long as the Python wrapper.
-THP_API PyObject * THPGenerator_initDefaultGenerator(at::Generator cdata);
+TORCH_PYTHON_API PyObject * THPGenerator_initDefaultGenerator(at::Generator cdata);
 
 #define THPGenerator_Check(obj) \
   PyObject_IsInstance(obj, THPGeneratorClass)
 
-THP_API PyObject *THPGeneratorClass;
+TORCH_PYTHON_API extern PyObject *THPGeneratorClass;
```

Comment:
Why do you need to re-add the extern here? (and in other places)

Reply:
`extern` is necessary for variables defined in another file, but it doesn't apply to classes or structs which is why  there were two macros: `THP_API` and `THP_CLASS`. Now we just have one and manually add `extern`. This is consistent with how all the other `_API` macros work.